spec/lib/pedicel/base_spec.rb

```diff
@@ -0,0 +1,425 @@
+        Pedicel.config[:oids][:leaf_certificate] = 'invalid oid'
+
+        is_expected.to raise_error(Pedicel::SignatureError, /no.*leaf.*found/)
+      end
+
+      it 'errs if there is no intermediate OID' do
+        Pedicel.config[:oids][:intermediate_certificate] = 'invalid oid'
+
+        is_expected.to raise_error(Pedicel::SignatureError, /no.*intermediate.*found/)
+      end
+
+      it 'errs if there are neither a leaf nor an intermediate OID' do
+        Pedicel.config[:oids][:leaf_certificate] = 'invalid oid'
+        Pedicel.config[:oids][:intermediate_certificate] = 'invalid oid'
+
+        is_expected.to raise_error(Pedicel::SignatureError, /no.*(leaf|intermediate).*found/)
+      end
+    end
+
+    context 'values' do
+      subject { Pedicel::Base.extract_certificates(signature: signature) }
+
+      it 'is truthy when all checks are good' do
+        is_expected.to be_truthy
+      end
+
+      it 'extracts leaf, intermediate, and root' do
+        is_expected.to eq([backend.leaf_certificate, backend.intermediate_certificate, backend.ca_certificate])
+      end
+    end
+  end
+
+  describe 'Pedicel::Base.extract_certificates' do
+    before (:each) { backend.encrypt_and_sign(token, recipient: client) }
+
+    let (:another_backend) { PedicelPay::Backend.generate }
+
+    let (:signature) { OpenSSL::PKCS7.new(Base64.strict_decode64(token.signature)) }
+
+    it 'errs if there are multiple leaf OIDs' do
+      signature.add_certificate(another_backend.leaf_certificate)
+
+      expect{Pedicel::Base.extract_certificates(signature: signature)}.to raise_error(Pedicel::SignatureError, /no unique leaf/)
+    end
+
+    it 'errs if there are multiple intermediate OIDs' do
+      signature.add_certificate(another_backend.intermediate_certificate)
+
+      expect{Pedicel::Base.extract_certificates(signature: signature)}.to raise_error(Pedicel::SignatureError, /no unique intermediate/)
+    end
+
+    it 'errs if there are multiple leaf and intermediate OIDs' do
+      signature.add_certificate(another_backend.leaf_certificate)
+      signature.add_certificate(another_backend.intermediate_certificate)
+
+      expect{Pedicel::Base.extract_certificates(signature: signature)}.to raise_error(Pedicel::SignatureError, /no unique (leaf|intermediate)/)
+    end
+
+    it 'errs if there are multiple certificates that are neither leaf nor intermediate' do
+      signature.add_certificate(another_backend.ca_certificate)
+
+      expect{Pedicel::Base.extract_certificates(signature: signature)}.to raise_error(Pedicel::SignatureError, /\Atoo many certificates found in the signature:/)
+    end
+  end
+
+  describe 'Pedicel::Base.verify_root_certificate' do
+    it 'does not err when the certificates are equal' do
+      expect{Pedicel::Base.verify_root_certificate(root: backend.ca_certificate, trusted_root: PedicelPay::Backend.generate.ca_certificate)}.to raise_error(Pedicel::SignatureError, /root.*not trusted/)
+    end
```

Comment:
A bit contradictory: `it 'does not err when ...` vs `expect{...}.to raise_error(...)`